### PR TITLE
[openshift-*] compare resources recursively by desired item

### PR DESCRIPTION
**Pull request by maorfr**
based on a suggestion by the glorious @jmelis:
https://gist.github.com/jmelis/594ca4fc7566159c4efbce8e80c4bc3f

compare resources only by the desired resource and the attributes it has. in case default values are added by k8s, they are ignored in this comparison.

### CODE REVIEW

File: reconcile/openshift_base.py [hunk 1]
```diff
@@ -232,6 +232,15 @@ def realize_data(dry_run, oc_map, ri,
                     ).format(cluster, namespace, resource_type, name)
                     logging.info(msg)
 
+                # don't apply if resources match
+                elif d_item == c_item:
```
Comment (apahim) [inline reconcile/openshift_base.py]: great start!

File: utils/openshift_resource.py [hunk 2]
```diff
@@ -28,6 +28,48 @@ def __init__(self, body, integration, integration_version,
...
+    def __eq__(self, other):
+        return self.obj_intersect_equal(self.body, other.body)
+
+    def obj_intersect_equal(self, obj1, obj2):
+        if obj1.__class__ != obj2.__class__:
+            return False
+        elif isinstance(obj1, dict):
+            for obj1_k, obj1_v in obj1.items():
+                obj2_v = obj2.get(obj1_k, None)
+                if not obj2_v:
```
Comment (apahim) [inline utils/openshift_resource.py]: `if obj2_v is not None` (so you don't enter the `if` when `obj2` is `0` or `''` )

File: utils/openshift_resource.py [hunk 3]
```diff
@@ -28,6 +28,48 @@ def __init__(self, body, integration, integration_version,
         self.error_details = error_details
         self.verify_valid_k8s_object()
 
+    def __eq__(self, other):
+        return self.obj_intersect_equal(self.body, other.body)
+
+    def obj_intersect_equal(self, obj1, obj2):
+        if obj1.__class__ != obj2.__class__:
+            return False
+        elif isinstance(obj1, dict):
```
Comment (apahim) [inline utils/openshift_resource.py]: just `if` her... no need for elif.

File: utils/openshift_resource.py [hunk 4]
```diff
@@ -28,6 +28,48 @@ def __init__(self, body, integration, integration_version,
...
+                obj2_v = obj2.get(obj1_k, None)
+                if not obj2_v:
+                    return False
+                if obj1_k == 'cpu':
+                    equal = self.cpu_equal(obj1_v, obj2_v)
+                    if not equal:
+                        return False
+                elif not self.obj_intersect_equal(obj1_v, obj2_v):
+                    return False
+        elif isinstance(obj1, list):
```
Comment (apahim) [inline utils/openshift_resource.py]: same here... just `if`

File: utils/openshift_resource.py [hunk 5]
```diff
@@ -28,6 +28,48 @@ def __init__(self, body, integration, integration_version,
...
+            if len(obj1) != len(obj2):
+                return False
+            for i in range(len(obj1)):
+                if not self.obj_intersect_equal(obj1[i], obj2[i]):
+                    return False
+        else:
```
Comment (apahim) [inline utils/openshift_resource.py]: it's safe to remove the else

File: reconcile/openshift_base.py (hunk 1 above)
Comment (maorfr) [reply reconcile/openshift_base.py]: thanks!

File: utils/openshift_resource.py (hunk 2 above)
Comment (maorfr) [reply utils/openshift_resource.py]: thanks!
look at what i did. does it makes sense?

File: utils/openshift_resource.py (hunk 3 above)
Comment (maorfr) [reply utils/openshift_resource.py]: thanks!

File: utils/openshift_resource.py (hunk 5 above)
Comment (maorfr) [reply utils/openshift_resource.py]: thanks!

File: utils/openshift_resource.py (hunk 4 above)
Comment (maorfr) [reply utils/openshift_resource.py]: thanks!

File: utils/openshift_resource.py (hunk 2 above)
Comment (apahim) [reply utils/openshift_resource.py]: yep, looks great. very easy to read.

**Pull request merged**